Fix/update acm asm

dkassab opened the pull request:
Updating documentation and fixing formatting issues.

daniel-cit commented:
Hi @dkassab Thanks for the PR
there are two lint errors:

- A missing newline
```
Step #1 - "lint-tests": Checking for missing newline at end of file
Step #1 - "lint-tests": Error: No newline at end of file ./6-anthos-install/README.md
```

- A missing required argument
```
Step #1 - "lint-tests": terraform_validate ./app-foundation/4-projects/shared 
Step #1 - "lint-tests": 
Step #1 - "lint-tests": Error: Missing required argument
Step #1 - "lint-tests": 
Step #1 - "lint-tests":   on infra_pipeline.tf line 44, in module "infra_pipelines":
Step #1 - "lint-tests":   44: module "infra_pipelines" {
Step #1 - "lint-tests": 
Step #1 - "lint-tests": The argument "impersonate_service_account" is required, but no definition was
Step #1 - "lint-tests": found
```

The missing required argument was fixed in PR https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/pull/40
It should be fixed if you rebase with the current version of main

daniel-cit reviewed (changes requested):
- There are some changes that need to be reverted,
which are the ones related to `gcr.io/bank-of-anthos`,
they are marked individually in the files. 
The text `gcr.io/bank-of-anthos` is replaced automatically in step 7.



- It would help the user if **all** the variables that depende in the output of previous steps
would be defined in the [define-required-environment-variables](https://github.com/dkassab/terraform-example-foundation-app/tree/fix/update-acm-asm/6-anthos-install#define-required-environment-variables)
section for example these variable cloud be move to the section:

```
export CICD_PROJECT_ID=YOUR_CICD_PROJECT_ID
```

and 

```
export SQL_PROJECT_ID=YOUR_SQL_PROJECT_ID
export SQL_INSTANCE_NAME_WEST=YOUR_SQL_INSTANCE_NAME_WEST
export SQL_INSTANCE_NAME_EAST=YOUR_SQL_INSTANCE_NAME_EAST
```


- We have three project IDs in the README.md for step 6 and
it would help the user if they were consistentelly named in the environement variable 
and in the replace tokens inside the yaml files
  - the GKE project ID: 
    - PROJECT_ID -> GKE_PROJECT_ID (env var)
    - PROJECT_ID -> GKE_PROJECT_ID (yaml)
  - the CI/CD project ID
    - PROJECT_ID -> CICD_PROJECT_ID (env var)
    - PROJECT_ID -> CICD_PROJECT_ID (yaml
  - the SQL project ID
    - SQL_PROJECT_ID -> SQL_PROJECT_ID (env var)(keep)

for example, this command at the end of the readme

```
gcloud iam service-accounts add-iam-policy-binding \
--role roles/iam.workloadIdentityUser \
--member "serviceAccount:$PROJECT_ID.svc.id.goog[transactions/transactions]" \
boa-gsa@$PROJECT_ID.iam.gserviceaccount.com
```

should be using 
- PROJECT_ID -> GKE_PROJECT_ID

Since the `boa-gsa` service account is create in the GKE project.

rutalreja-deloitte commented:
@googlebot I consent.

rutalreja-deloitte commented on 6-anthos-install/acm-repos/accounts/userservice.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: userservice
-          image: gcr.io/bank-of-anthos/userservice:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/userservice:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/frontend/frontend.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: front
-          image: gcr.io/bank-of-anthos/frontend:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/frontend:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/frontend/loadgenerator.yaml:
```diff
@@ -33,7 +33,7 @@ spec:
       containers:
         - name: loadgenerator
-          image: gcr.io/bank-of-anthos/loadgenerator:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/loadgenerator:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/transactions/balancereader.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: balancereader
-          image: gcr.io/bank-of-anthos/balancereader:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/balancereader:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/transactions/ledgerwriter.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: ledgerwriter
-          image: gcr.io/bank-of-anthos/ledgerwriter:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/ledgerwriter:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/transactions/transactionhistory.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: transactionhistory
-          image: gcr.io/bank-of-anthos/transactionhistory:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/transactionhistory:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/acm-repos/accounts/contacts.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: contacts
-          image: gcr.io/bank-of-anthos/contacts:v0.4.3
+          image: us-east1-docker.pkg.dev/prj-bu1-s-app-cicd-cfdd/prj-bu1-s-app-cicd-cfdd-boa-image-repo/contacts:v0.4.3
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +229,259 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
 ### Populate the CSR repos
 For configuring and deploying the applicaiton, we are using multi-repo mode in ACM. This mode allows syncing from multiple repositories. In this excample, we have one root repository that hosts the cluster-wide and namespace-scoped configurations, and three namespace repositories to host the application manifests.
 
+Find the Project ID for your CI/CD project. It will look something like this: "prj-bu1-s-app-cicd-[RANDOM]"
```
fixed in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89

rutalreja-deloitte commented:
Fixes in c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89
- Changes requested in @daniel-cit 's comment
- Remove --cluster-labels not needed
- Add missing instruction for frontend-sa.yaml

rutalreja-deloitte commented:
@dkassab and @daniel-cit 
Noticed missing instructions for 
- [acm root-sync yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-configs/root-sync.yaml#L24) in ReadME, line should also have GKE_PROJECT_ID instead of PROJECT_ID
- [cloud-sql-admin-secret yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-repos/root-config-repo/namespaces/boa/accounts/cloud-sql-admin-secret.yaml#L17) instruction to update this in ReadME
- [app-secret-front yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-repos/root-config-repo/namespaces/boa/frontend/app-secrets-front.yaml#L23) instruction to update this in ReadME as well as the correct project env variable GKE most likely

dkassab commented:
> @dkassab and @daniel-cit
> Noticed missing instructions for
> 
> * [acm root-sync yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-configs/root-sync.yaml#L24) in ReadME, line should also have GKE_PROJECT_ID instead of PROJECT_ID
> * [cloud-sql-admin-secret yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-repos/root-config-repo/namespaces/boa/accounts/cloud-sql-admin-secret.yaml#L17) instruction to update this in ReadME
> * [app-secret-front yaml](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-repos/root-config-repo/namespaces/boa/frontend/app-secrets-front.yaml#L23) instruction to update this in ReadME as well as the correct project env variable GKE most likely
> 
> P.S Also a lot of files in the `6-anthos-install/acm-repos/root-config-repo/namespaces/boa/` repo folders are entirely commented out, maybe some instructions in the README are required to inform user what role the files play and at what stage

- I fixed acm root-sync.yaml file.
- cloud-sql-admin-secret is not needed anymore, I removed it.
- This secret is not needed for this phase, but will need it once we implement secret manager and CSI secret driver.

rutalreja-deloitte commented:
Fixes #51 
- Exposed `bastion_members` in `5-infra/env/main.tf` and added supporting instruction in 6-README
- Added missing Readme instruction for [acm-repo-sync](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-configs/root-sync.yaml#L24)
- Commented out [app-secret-front](https://github.com/GoogleCloudPlatform/terraform-example-foundation-app/blob/c599738f5cdeaa8ff6f3c5f8a09acb3b16233e89/6-anthos-install/acm-repos/root-config-repo/namespaces/boa/frontend/app-secrets-front.yaml#L23)

rutalreja-deloitte commented:
LGTM
@daniel-cit and @bharathkkb please review when you get a chance

daniel-cit commented on 6-anthos-install/README.md:
````diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+1. create private key secret in "accounts", "transactions" and "fronted" namespaces.
+    ```
+    # On Cluster 1
+    kubectl create secret generic git-creds --namespace=transactions --context ${CTX_1} --from-file=ssh="${HOME}/.ssh/id_rsa"
````
I got 
```
error: failed to create secret namespaces "transactions" not found
```

If I retry it works:

```
secret/git-creds created
```

Do we have to wait for something to complete before creating the secret?

daniel-cit commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+    kubectl create secret generic git-creds --namespace=frontend --context ${CTX_1} --from-file=ssh="${HOME}/.ssh/id_rsa"
+
+    # On Cluster 2
+    kubectl create secret generic git-creds --namespace=transactions --context ${CTX_2} --from-file=ssh="${HOME}/.ssh/id_rsa"
```
also failed in the first execution
```
error: failed to create secret namespaces "transactions" not found
```


It works after fixing 

```
kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-west.yaml
```

daniel-cit commented on 6-anthos-install/README.md:
````diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+    ```
+    kubectl apply --context=${CTX_1} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/root-sync.yaml
+
+    kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/root-sync.yaml
````
I got the error

```
 kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/root-sync.yaml
error: unable to recognize "/home/ext_REDACTED_USER_example_com/terraform-example-foundation-app/6-anthos-install/acm-configs/root-sync.yaml": no matches for kind "RootSync" in version "configsync.gke.io/v1alpha1"
```

IT works after fixing 

```
kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-west.yaml
```

daniel-cit commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+    # On Cluster 2
+    kubectl create secret generic git-creds --namespace=transactions --context ${CTX_2} --from-file=ssh="${HOME}/.ssh/id_rsa"
+
+    kubectl create secret generic git-creds --namespace=accounts --context ${CTX_2} --from-file=ssh="${HOME}/.ssh/id_rsa"
```
failed with 
```
error: failed to create secret namespaces "accounts" not found
```

It works after fixing 

```
kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-west.yaml
```

daniel-cit commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+
+    kubectl create secret generic git-creds --namespace=accounts --context ${CTX_2} --from-file=ssh="${HOME}/.ssh/id_rsa"
+
+    kubectl create secret generic git-creds --namespace=frontend --context ${CTX_2} --from-file=ssh="${HOME}/.ssh/id_rsa"
```
failed with 
```
error: failed to create secret namespaces "frontend" not found
```

It works after fixing 

```
kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-west.yaml
```

daniel-cit commented on 6-anthos-install/README.md:
````diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
-    ```
+    kubectl apply --context=${CTX_1} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-east.yaml
+
+    kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-config/config-management-west.yaml
````
```suggestion
    kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/acm-configs/config-management-west.yaml
```

daniel-cit commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
+    kubectl apply -n transactions --context ${CTX_1} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/db-scripts/populate-ledger-db.yaml
 
-    kubectl apply --context=${CTX_2} -f ${HOME}/terraform-example-foundation-app/acm-config/root-sync.yaml
+    kubectl apply -n transactions --context ${CTX_2} -f ${HOME}/terraform-example-foundation-app/6-anthos-install/db-scripts/populate-ledger-db.yaml
```
The file ${HOME}/terraform-example-foundation-app/6-anthos-install/db-scripts/populate-ledger-db.yaml does not exist

daniel-cit reviewed (changes requested):
This version has many improvements but there are still a few issues to fix

daniel-cit commented on 6-anthos-install/README.md:
```diff
@@ -215,132 +231,263 @@ With MCI, we need to select a cluster to be the configuration cluster. In this c
         port: 15020
         requestPath: /healthz/ready
```
```suggestion
        port: 8080
        requestPath: /ready
```

bharathkkb commented on 6-anthos-install/README.md:
````diff
@@ -1,146 +1,159 @@
 1. Creates a secret that grants access to the Kube API Server for cluster 1
-    ```console
+    ```
     ./istioctl x create-remote-secret \
````
It maybe valuable to keep `./`. If the user has an older version of istioctl in PATH then it may error out with these configs
